server/channels/store/storetest/post_store.go

```diff
@@ -4015,8 +4015,9 @@ func testPostStorePermanentDeleteBatch(t *testing.T, rctx request.CTX, ss store.
 	require.Equal(t, 1, len(rows))
 	require.Equal(t, 2, len(rows[0].Ids))
 	// Clean up retention ids table
-	err = ss.Reaction().DeleteOrphanedRowsByIds(rows[0])
+	deleted, err = ss.Reaction().DeleteOrphanedRowsByIds(rows[0])
 	require.NoError(t, err)
+	require.Equal(t, int64(0), deleted)
```

Comment:
Note to self: Why is this cleanup needed, if no reaction get deleted?

Reply:
This call is need to clean the post from RetentionIdsForDeletion by calling https://github.com/mattermost/mattermost/blob/a3ae02e266172fdf48e63890eca9c734291f85ca/server/channels/store/sqlstore/retention_policy_store.go#L1155-L1162